src/App.tsx

```diff
@@ -16,35 +26,121 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([...goodsFromServer]);
+  const [sortState, setSortState] = useState<{
+    reversed: boolean;
+    criterion: SortType;
+  }>({
+    reversed: false,
+    criterion: SortType.None,
+  });
+  const [isReset, setIsReset] = useState(false as boolean);
+
+  const sortGoodsByAlphabet: ClickButton = () => {
+    const sortedGoods: Good[] = [...goods].sort((a, b) => {
+      if (sortState.reversed) {
+        return b.localeCompare(a);
+      }
+
+      return a.localeCompare(b);
+    });
+
+    setGoods(sortedGoods);
+    setSortState({ ...sortState, criterion: SortType.Alphabet });
+    setIsReset(true);
+  };
+
+  const sortGoodsByLength: ClickButton = () => {
+    const sortedGoods = [...goods].sort((a, b) => {
+      if (sortState.reversed) {
+        return b.length - a.length;
+      }
+
+      return a.length - b.length;
+    });
+
+    setGoods(sortedGoods);
+    setSortState({ ...sortState, criterion: SortType.Length });
+    setIsReset(true);
+  };
+
+  const reverseGoods: ClickButton = () => {
```

Comment:
Try to simplify, prepared goods just store in a variable and create a function that prepare them
```typescript
const preparedGoods = prepareGoods(goods, sortState);
```